packages/framework/tests/Unit/Component/Doctrine/QueryBuilderExtenderTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrameworkBundle\Unit\Component\Doctrine;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\QueryBuilder;
+use PHPUnit\Framework\TestCase;
+use Shopsys\FrameworkBundle\Component\Doctrine\QueryBuilderExtender;
+use Shopsys\FrameworkBundle\Component\EntityExtension\EntityNameResolver;
+use Shopsys\FrameworkBundle\Model\Category\Category;
+use Shopsys\FrameworkBundle\Model\Product\Product as BaseProduct;
+use Shopsys\ShopBundle\Model\Product\Product;
```

Comment:
Sadly this will not work from `shopsys/framework`.

I would suggest to create new extended class of `Product` as a fixture for tests only. The similar way how it is done for testing elasticsearch https://github.com/shopsys/shopsys/blob/master/packages/framework/tests/Unit/Component/Elasticsearch/__fixtures/CategoryIndex.php